mantidimaging/core/data/dataset.py

```diff
@@ -25,11 +25,14 @@ def remove_nones(image_stacks: list[ImageStack | None]) -> list[ImageStack]:
 
 class BaseDataset:
 
-    def __init__(self, *, name: str = ""):
+    def __init__(self, *, name: str = "", stacks: list[ImageStack] | None = None) -> None:
         self._id: uuid.UUID = uuid.uuid4()
-        self._recons = ReconList()
         self.name = name
+
+        self._recons = ReconList()
         self._sinograms: ImageStack | None = None
+        stacks = [] if stacks is None else stacks
```

Comment:
If i'm understanding this correctly, could we set the default value of stacks in the initialiser as an empty list instead of None to save us needing to do this if no argument is passed?

Reply:
If there is a chance we could be calling with None, does this need to be the case or can it be avoided to allow for just setting a default return value of an empty list?

Reply:
We can't use an empty list as the default argument as it is mutable and would end up getting shared by multiple instances. ( https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments )